### PR TITLE
Fix PHP 8 compatibility in 2.x

### PR DESCRIPTION
**Pull request by Tobion**
My library fails with php 8 using swagger 2.x, see https://github.com/Tobion/OpenAPI-Symfony-Routing/pull/7
This is due to the namespace not being parsed correctly.

This PR fixes PHP 8 compatibility.
It would be very hard to test with php 5.6 until php 8 due to phpunit bc breaks and incompatibilities, e.g. phpunit <9 will not support php 8. so just droppping unsupported php versions is the best choice which was also done in 3.x (master).

This backports #830, #837, 29e913a, parts of 3f1cee9 to fix php 8 errors and tests.

### CODE REVIEW

**Comment by Tobion** [comment]
@DerManoMann tests pass. I would appreciate if you tag this as 2.1 release.

File: composer.json [hunk 1]
```diff
@@ -33,13 +33,13 @@
...
-        "zendframework/zend-form": "<2.8",
         "squizlabs/php_codesniffer": ">=2.7",
-        "phpunit/phpunit": ">=4.8.35 <=5.6"
+        "phpunit/phpunit": "^8 || ^9"
     },
     "autoload-dev": {
         "psr-4": {
-            "SwaggerTests\\": "tests/"
+            "SwaggerTests\\": "tests/",
+            "AnotherNamespace\\": "tests/Fixtures/AnotherNamespace"
```
Comment (DerManoMann) [inline composer.json:42]: Noit sure if this is really needed seeing that the fixtures do not properly use PSR-4 namespaces.

**Review by DerManoMann — APPROVED** [review]
Nice work - thanks.

File: composer.json (hunk 1 above)
Comment (Tobion) [reply composer.json:42]: The fixtures use psr-4 namespace autoloading corrctly. Just with strange namespaces (\SwaggerTests, \AnotherNamespace). So I think this is needed to load the annotation examples.

Comment (DerManoMann) [reply composer.json:42]: Hmm, I would argue that
```
namespace SwaggerFixtures;
```
is not a correct namespace according to PSR-0/PSR-4 as it does not match the folder name at all.

Nevertheless, that's not really important for the fixtures so all is good.

**Comment by DerManoMann** [comment]
@bfanger @zircote I agree with @Tobion that a new tag/release for this would be nice.... As suggested, `2.1.0` seems like the right thing

**Comment by Tobion** [comment]
@DerManoMann can't you create the tag?

**Comment by DerManoMann** [comment]
I'll double check - looks like I can... I always mix up GH and packagist permissions :/

**Comment by Tobion** [comment]
@DerManoMann thanks for the tag. Unfortunately the release is not visible to packagist: https://packagist.org/packages/zircote/swagger-php. Can you trigger a refresh?
And also enable auto-update: https://packagist.org/about#how-to-update-packages

**Comment by DerManoMann** [comment]
damn, got distracted! autoupdate is definitely something I cannot configure

**Pull request merged**